addons/sale/models/account_invoice.py

```diff
@@ -97,3 +97,8 @@ def _get_invoice_delivery_partner_id(self):
         # OVERRIDE
         self.ensure_one()
         return self.partner_shipping_id.id or super(AccountMove, self)._get_invoice_delivery_partner_id()
+
+    def _is_downpayment(self):
+        # OVERRIDE
+        self.ensure_one()
+        return bool(self.line_ids.sale_line_ids) and all(sale_line.is_downpayment for sale_line in self.line_ids.sale_line_ids)
```

Comment:
@dbkosky please avoid bool if not necessary

Reply:
bool is necessary, otherwise it returns `sale.order.line()`, the return type should be bool

I tried without bool before, and comparison in the `_l10n_it_get_document_type` method failed for all scenarios where the document is not a downpayment.
`invoice._is_downpayment() == infos.get('downpayment', False),`
since `sale.order.line() == False` is False.

Reply:
Maybe you just need or False in the end